go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -368,6 +373,11 @@ func (dp diffPartition) getRowIter(ctx *sql.Context, ddb *doltdb.DoltDB, ss *sch
 	toCmInfo := commitInfo{types.String(dp.toName), dp.toDate, toCol.Tag, toDateCol.Tag}
 
 	rd := diff.NewRowDiffer(ctx, fromSch, toSch, 1024)
+	if _, ok := rd.(*diff.EmptyRowDiffer); ok {
+		ctx.Warn(PrimaryKeyChanceWarningCode, fmt.Sprintf(PrimaryKeyChangeWarning, dp.fromName, dp.toName))
+		return sql.RowsToRowIter(), nil
+	}
+
```

Comment:
Why do we check this here? Where do we return `EmptyRowDIffer`?

Reply:
We return empty row differ async_differ.go when we call `NewRowDiffer`. We check to primarily throw a warning for users. The emptyRowDiffer is useful in reducing the conditional logic checks of schema changes since it just returns an empty diff. Cc. dolt constraints verify

Reply:
My reading of this is that `isDiffablePartition()` will prevent us from ever creating an `async_differ` for two roots that can't be diffed due to incompatible PK sets. Is this not true? 
 We definitely want to catch these cases earlier, higher up the stack